### PR TITLE
Handle empty weibars in EthereumTransaction

### LINKED ISSUE #4025: Error when parsing zero length EthereumTransaction maxFeePerGas
### Description

The importer got an error `java.lang.NumberFormatException: Zero length BigInteger` when parsing an EthereumTransaction.

### Steps to reproduce

Send an EthereumTransaction with maxFeePerGas to an empty byte array.

### Additional context

```java
Jun 29 16:56:14 mirrornode-preview-testnet-parser-1 java[12802]: 2022-06-29T10:56:14.911-0600 ERROR scheduling-5 c.h.m.i.p.r.RecordFileParser Error parsing file 2022-06-29T14_29_25.169327431Z.rcd after 1.195 ms java.lang.NumberFormatException: Zero length BigInteger
        at java.base/java.math.BigInteger.<init>(BigInteger.java:310)
        at java.base/java.math.BigInteger.<init>(BigInteger.java:338)
        at com.hedera.mirror.common.converter.WeiBarTinyBarConverter.weiBarToTinyBar(WeiBarTinyBarConverter.java:31)
        at com.hedera.mirror.importer.parser.record.transactionhandler.EthereumTransactionHandler.convertGasWeiToTinyBars(EthereumTransactionHandler.java:124)
        at com.hedera.mirror.importer.parser.record.transactionhandler.EthereumTransactionHandler.updateTransaction(EthereumTransactionHandler.java:86)
        at com.hedera.mirror.importer.parser.record.entity.EntityRecordItemListener.onItem(EntityRecordItemListener.java:177)
        at com.hedera.mirror.importer.parser.record.entity.EntityRecordItemListener.onItem(EntityRecordItemListener.java:112)
        at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:196)
        at reactor.core.publisher.FluxFilterFuseable$FilterFuseableSubscriber.tryOnNext(FluxFilterFuseable.java:148)
        at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.tryOnNext(FluxPeekFuseable.java:530)
        at reactor.core.publisher.FluxIterable$IterableSubscriptionConditional.fastPath(FluxIterable.java:644)
        at reactor.core.publisher.FluxIterable$IterableSubscriptionConditional.request(FluxIterable.java:529)
        at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.request(FluxPeekFuseable.java:437)
        at reactor.core.publisher.FluxFilterFuseable$FilterFuseableSubscriber.request(FluxFilterFuseable.java:191)
        at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.request(FluxPeekFuseable.java:144)
        at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.request(FluxPeekFuseable.java:144)
        at reactor.core.publisher.MonoCount$CountSubscriber.onSubscribe(MonoCount.java:81)
        at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onSubscribe(FluxPeekFuseable.java:178)
        at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onSubscribe(FluxPeekFuseable.java:178)
        at reactor.core.publisher.FluxFilterFuseable$FilterFuseableSubscriber.onSubscribe(FluxFilterFuseable.java:87)
        at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.onSubscribe(FluxPeekFuseable.java:471)
        at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:161)
        at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:87)
        at reactor.core.publisher.Mono.subscribe(Mono.java:4400)
        at reactor.core.publisher.Mono.block(Mono.java:1706)
        at com.hedera.mirror.importer.parser.record.RecordFileParser.doParse(RecordFileParser.java:133)
        at com.hedera.mirror.importer.parser.record.RecordFileParser.doParse(RecordFileParser.java:48)
        at com.hedera.mirror.importer.parser.AbstractStreamFileParser.parse(AbstractStreamFileParser.java:79)
        at com.hedera.mirror.importer.parser.record.RecordFileParser.parse(RecordFileParser.java:111)
        at com.hedera.mirror.importer.parser.record.RecordFileParser.parse(RecordFileParser.java:48)
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$FastClassBySpringCGLIB$$8b9310db.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
        at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:97)
        at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)
        at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)
        at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:133)
        at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:160)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$EnhancerBySpringCGLIB$$be09a22e.parse(<generated>)
        at com.hedera.mirror.importer.config.MessagingConfiguration.lambda$integrationFlow$1(MessagingConfiguration.java:107)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.springframework.integration.handler.LambdaMessageProcessor.processMessage(LambdaMessageProcessor.java:104)
        at org.springframework.integration.handler.ServiceActivatingHandler.handleRequestMessage(ServiceActivatingHandler.java:105)
        at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:136)
        at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:56)
        at org.springframework.integration.endpoint.PollingConsumer.handleMessage(PollingConsumer.java:158)
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.messageReceived(AbstractPollingEndpoint.java:475)
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:461)
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:413)
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$4(AbstractPollingEndpoint.java:348)
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:57)
        at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.execute(ErrorHandlingTaskExecutor.java:55)
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$5(AbstractPollingEndpoint.java:341)
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:95)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
        Suppressed: java.lang.Exception: #block terminated with an error
                at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:99)
                at reactor.core.publisher.Mono.block(Mono.java:1707)
                ... 50 more
```

### Hedera network

previewnet

### Version

v0.59.0

### Operating system

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:
* Fix handling of empty weibars for `EthereumTransaction`
* Fix handling of unsigned big integers for `EthereumTransaction`

**Related issue(s)**:

Fixes #4025

**Notes for reviewer**:
Also verified it works with actual problem RCD from previewnet

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/converter/WeiBarTinyBarConverter.java [hunk 1]
```diff
@@ -23,12 +23,19 @@
...
 
 public class WeiBarTinyBarConverter {
+
     public static final WeiBarTinyBarConverter INSTANCE = new WeiBarTinyBarConverter();
     public static final Long WEIBARS_TO_TINYBARS = 10_000_000_000L;
     public static final BigInteger WEIBARS_TO_TINYBARS_BIGINT = BigInteger.valueOf(WEIBARS_TO_TINYBARS);
 
     public byte[] weiBarToTinyBar(byte[] weibar) {
-        return weibar == null ? null : new BigInteger(weibar).divide(WEIBARS_TO_TINYBARS_BIGINT).toByteArray();
+        if (weibar == null || weibar.length == 0) {
```
Comment (Nana-EC) [inline hedera-mirror-common/src/main/java/com/hedera/mirror/common/converter/WeiBarTinyBarConverter.java:32]: nit:
```suggestion
        if (ArrayUtils.isEmpty(weibar) {
```

**Review by Nana-EC — APPROVED** [review]
LG. Minor suggestion if you circle back

**Review by xin-hedera — APPROVED** [review]

**Review by edwin-greene — APPROVED** [review]

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/converter/WeiBarTinyBarConverter.java (hunk 1 above)
Comment (steven-sheehy) [reply hedera-mirror-common/src/main/java/com/hedera/mirror/common/converter/WeiBarTinyBarConverter.java]: Updated

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**